### PR TITLE
Migrate to TypeScript

### LINKED ISSUE #73: Migrate to TypeScript
### What feature are you adding?

Migrating the codebase to TypeScript

### Describe your issue

As was the plan during the open source challenge, we intended to migrate the codebase to TypeScript after the challenge.

### PR DESCRIPTION
**Pull request by antosan**
## Fixes Issue

- closes #73 

## Changes proposed

As was the plan during the open source challenge, we intended to migrate the codebase to TypeScript after the challenge. This PR implements this migration.

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Note to reviewers

- [ ] Run `npm run lint` - there should be no linter warnings or errors
- [ ] Run `npm run test` - the available basic test should pass
- [ ] Run `npm run prettier` - all files should already be formatted according to the rules
- [ ] Run `npm run typecheck` - there should be no TypeScript errors
- [ ] Run `npm run build` - the build should be successful
- [ ] Run `npm run dev` - the project should run without any errors

I deleted the empty `*.spec.jsx` and `*.stories.jsx` files. The setup for these is still in the project and we should add them back as we write the tests and stories.

### CODE REVIEW

File: src/components/Banner/Banner.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 
-const Banner = () => {
+export default function Banner() {
```
Comment (orama254) [inline src/components/Banner/Banner.tsx:3]: @antosan is this a linter based change or a stylistic preference?
(the function declaration)

Comment (antosan) [reply src/components/Banner/Banner.tsx:3]: @orama254 This is not part of the linter. I checked the components in the project and the majority were implemented as function declarations. So while adding types for the props across the project, I opted to make them uniform.

Comment (orama254) [reply src/components/Banner/Banner.tsx:3]: noted.

**Review by orama254 — APPROVED** [review]
Looks Good To Me ✅

File: src/pages/events/index.page.tsx [hunk 2]
```diff
@@ -2,8 +2,22 @@ import React from 'react';
 import DisplayRC from '../../components/EventsDisplay';
 import logo from '../../../public/reactdevske.svg';
 import Head from 'next/head';
+import type { GetStaticProps } from 'next';
 
-export const getStaticProps = async () => {
+type EventData = {
```
Comment (Lekipising) [inline src/pages/events/index.page.tsx:7]: This interface has been defined in `src/components/EventsDisplay/DisplayRC.tsx`. Perhaps we can create a shared interface and import to all the files using it

Comment (antosan) [reply src/pages/events/index.page.tsx]: @Lekipising Nice observation. Fixed in https://github.com/reactdeveloperske/reactdevske-website/pull/74/commits/760edb9682f1eed758952861e243f92906d6fe6d

Comment (Lekipising) [reply src/pages/events/index.page.tsx]: Awesome @antosan :ok_hand:

**Review by Lekipising — APPROVED** [review]

**Pull request merged**